components/api_request_helper/api_request_helper_unittest.cc

```diff
@@ -198,4 +211,47 @@ TEST_F(ApiRequestHelperUnitTest, EnableCache) {
               base::NullCallback(), true);
 }
 
+TEST_F(ApiRequestHelperUnitTest, JsonParsing) {
+  base::RunLoop run_loop;
+  SendMessage(base::BindRepeating(
+                  [](base::RunLoop* run_loop, const std::string& response) {
+                    EXPECT_EQ(" Hello there!", response);
+                    run_loop->Quit();
+                  },
+                  &run_loop),
+              "data: {\"completion\": \" Hello there!\", \"stop\": null}");
+  run_loop.Run();
+
+  base::RunLoop run_loop2;
+  SendMessage(
+      base::BindRepeating(
+          [](base::RunLoop* run_loop, const std::string& response) {
+            EXPECT_EQ(" Hello there! How are you?", response);
+            run_loop->Quit();
+          },
+          &run_loop2),
+      "data: {\"completion\": \" Hello there! How are you?\", \"stop\": null}");
+  run_loop2.Run();
+
+  // This test verifies that the callback is not called when the response is
+  // "[DONE]". We use a run loop to wait for the callback to be called, and we
+  // expect it to never be called.
```

Comment:
That's fine for now as it works for the 1 user of this API (AI Chat), but technically it should return the string "[DONE]" as far as I understand.

Reply:
Added a test to achieve this